test/solidity/Periphery/ReceiverChainflip.t.sol

```diff
@@ -0,0 +1,395 @@
+    bytes32 guid = bytes32("12345");
+    address receiverAddress = USER_RECEIVER;
+    address constant CHAINFLIP_NATIVE_ADDRESS =
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    Executor executor;
+    ERC20Proxy erc20Proxy;
+    address chainflipVault;
+
+    event ExecutorSet(address indexed executor);
+
+    function setUp() public {
+        customBlockNumberForForking = 18277082;
+        initTestBase();
+
+        chainflipVault = getConfigAddressFromPath(
+            "chainflip.json",
+            ".chainflipVault.mainnet"
+        );
+        vm.label(chainflipVault, "Chainflip Vault");
+
+        erc20Proxy = new ERC20Proxy(address(this));
+        executor = new Executor(address(erc20Proxy), address(this));
+        receiver = new ReceiverChainflip(
+            address(this),
+            address(executor),
+            chainflipVault
+        );
+        vm.label(address(receiver), "ReceiverChainflip");
+        vm.label(address(executor), "Executor");
+        vm.label(address(erc20Proxy), "ERC20Proxy");
+    }
+
+    function test_ContractIsSetUpCorrectly() public {
+        receiver = new ReceiverChainflip(
+            address(this),
+            address(executor),
+            chainflipVault
+        );
+
+        assertEq(address(receiver.executor()) == address(executor), true);
+        assertEq(receiver.chainflipVault() == chainflipVault, true);
+    }
+
+    function testRevert_OnlyChainflipVaultCanCallCfReceive() public {
+        // mock-send bridged funds to receiver contract
+        deal(ADDRESS_USDC, address(receiver), defaultUSDCAmount);
+
+        // call from deployer of ReceiverChainflip
+        vm.startPrank(address(this));
+        vm.expectRevert(UnAuthorized.selector);
+        receiver.cfReceive(
+            1, // srcChain (Ethereum)
+            abi.encodePacked(address(0)),
+            abi.encode("payload"),
+            ADDRESS_USDC,
+            defaultUSDCAmount
+        );
+        vm.stopPrank();
+
+        // call from random user
+        vm.startPrank(USER_SENDER);
+        vm.expectRevert(UnAuthorized.selector);
+        receiver.cfReceive(
+            1, // srcChain (Ethereum)
+            abi.encodePacked(address(0)),
+            abi.encode("payload"),
+            ADDRESS_USDC,
+            defaultUSDCAmount
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanDecodeChainflipPayloadAndExecuteSwapERC20() public {
+        // mock-send bridged funds to receiver contract
+        deal(ADDRESS_USDC, address(receiver), defaultUSDCAmount);
+
+        // Store initial balance
+        uint256 initialReceiverDAI = dai.balanceOf(receiverAddress);
+
+        // encode payload with mock data
+        (
+            bytes memory payload,
+            uint256 amountOutMin
+        ) = _getValidChainflipPayload(ADDRESS_USDC, ADDRESS_DAI);
+
+        // fake a call from Chainflip vault
+        vm.startPrank(chainflipVault);
+
+        vm.expectEmit();
```

Comment:
please use 
```vm.expectEmit(true,true,true,true,address(receiver);```

cause otherwise it's not checked if the event parameters actually match